package/gluon-core/luasrc/usr/lib/lua/gluon/wireless.lua

```diff
@@ -2,16 +2,45 @@ local sysconfig = require 'gluon.sysconfig'
 local site = require 'gluon.site'
 local util = require 'gluon.util'
 
+local unistd = require 'posix.unistd'
+local dirent = require 'posix.dirent'
+
 
 local M = {}
 
 local function find_phy_by_path(path)
-	local phy = util.glob('/sys/devices/' .. path .. '/ieee80211/phy*')[1]
-		or util.glob('/sys/devices/platform/' .. path .. '/ieee80211/phy*')[1]
+	local device_path, phy_offset
+	-- Special handling required for multi-phy devices
+	if string.match(path, '+%d$') then
+		device_path, phy_offset = string.match(path, "^(.+)%+(%d+)$")
+
+		-- The UCI radio path is split into the path and a PHY offset,
+		-- which is incremented for each additional PHY available on
+		-- the device.
+	else
+		device_path = path
+		phy_offset = 0
+	end
+
+	-- Find the device path. Either it's located at /sys/devices or /sys/devices/platform
+	local path_prefix = ''
+	if not unistd.access('/sys/devices/' .. device_path .. '/ieee80211') then
+		path_prefix = 'platform/'
+	end
+
+	-- Get all available PHYs of the device and dertermine the one with the lowest index
+	local phy_names = dirent.dir('/sys/devices/' .. path_prefix .. device_path .. '/ieee80211')
+	local device_phy_idxs = {}
+	for _, v in ipairs(phy_names) do
+		local phy_idx = v:match('(%d)$')
+		phy_idx = tonumber(phy_idx)
```

Comment:
I think this should match `^phy(%d+)$` and check for nil - the iwinfo code also ignores entries that don't start with `phy`.

It is unclear to me how reliable the iwinfo code is, as it doesn't sort the entries returned by readdir. Maybe that's fine for sysfs?

Reply:
I can't commit on that, but it might have been built with less edge-cases in mind.

Reply:
Ah, I misunderstood the code of iwinfo, not sorting the entries is fine there.